fornax-hea/build-sas.sh

```diff
@@ -0,0 +1,194 @@
+export SAS_PYTHON=$ENV_DIR/sas/bin/python
+
+# This is where we will be putting the calibration files for XMM
+export SAS_CCF=$SUPPORT_DATA_DIR/xmm_sas/
+###########################################################
+
+
+##################### DOWNLOADING SAS #####################
+# SAS will be downloaded from HEASARC - this is the base for the populated URL
+base_sas_link=https://heasarc.gsfc.nasa.gov/FTP/xmm/software/sas/${sas_version}/Linux/Ubuntu${ubuntu_version}/
+
+# This is the structure of the software tar file name
+sas_file=sas_${sas_version}-Ubuntu${ubuntu_version}.tgz
+
+# Assembling the download link
+sas_link=$base_sas_link${sas_file}
+###########################################################
+
+
+########### Download and unpack required files ############
+wget $sas_link \
+	&& tar xvf $sas_file \
+	&& rm -f $sas_file
+###########################################################
+
+
+########## Determine name of install directory  ###########
+# Set the pattern to search for
+pattern="xmmsas_${sas_version}-*bin*"
+
+# Find all matching files and store them in an array
+matches=($pattern)
+
+# Get the number of matches
+num_matches=${#matches[@]}
+
+# Check the conditions
+# The first part checks if there are no matches, or if there is exactly one match
+# AND that match is the pattern itself (meaning no files were found).
+if [[ $num_matches -eq 0 ]] || [[ $num_matches -eq 1 && "${matches[0]}" == "$pattern" ]]; then
+  echo "Error: No matching files found for '$pattern'." >&2
+  exit 1
+elif [[ $num_matches -gt 1 ]]; then
+  echo "Error: Multiple matching files found for '$pattern'." >&2
+  exit 1
+else
+  # A single, real match was found, so we split the file name to retrieve the installation folder
+  sas_bin_file="${matches[0]}"
+  sas_install_dir="${sas_bin_file%%-bin*}"
+fi
+###########################################################
+
+
+############# Setup the SAS Conda environment #############
+# Creates a Conda definition file that can be used to setup the environment that SAS
+#  will be associated with
+# We use the Conda-hosted HEASoft to avoid downloading and building HEASoft manually
+cat <<EOF > conda-sas.yml
+name: sas
+channels:
+  - https://heasarc.gsfc.nasa.gov/FTP/software/conda
+  - conda-forge
+dependencies:
+  - python=$py_version
+  - pytest
+  - xorg-libx11
+  - xorg-libxext
+  - xorg-libsm
+EOF
+
+# Use the yml to create the SAS env
+bash /usr/local/bin/conda-env-install.sh
+###########################################################
+
+
+############ Installing SAS Python requirements ###########
+# Use the Python requirements file included in the SAS directory to install
+#  the module that it wants for some included Python bits (e.g. pySAS)
+# We remove the pyDS9 requirement - the package is archived, and making it on Fornax would be a waste of time
+sed -i '/pyds9/d' sas_python_packages.txt
+micromamba run -n sas pip install -r sas_python_packages.txt --no-cache-dir
```

Comment:
Since we are modifying the environment, we need to update the lock file:
```sh
micromamba env -n sas export > $ENV_DIR/sas/sas-lock.yml
cp $ENV_DIR/sas/sas-lock.yml $LOCK_DIR
```